install/resources/upgrade41.php

```diff
@@ -35,6 +35,28 @@ function upgrade41_dbchanges()
 	$db->insert_query('spiders', array("name" => "DuckDuckGo", "useragent" => "DuckDuckBot/1.1; (+http://duckduckgo.com/duckduckbot.html)"));
 	$db->insert_query('spiders', array("name" => "UptimeRobot", "useragent" => "Mozilla/5.0+(compatible; UptimeRobot/2.0; http://www.uptimerobot.com/)"));
 
+	// Iterate through moderator log and remove slashes
+	for ($x = 1; $x < $db->num_rows('moderatorlog'); $x++)
+	{
+
+		$moderatorlogitemarray = $db->simple_select("moderatorlog", "*", "tid='".$x."'", array(
```

Comment:
This is not efficient as it queries the db for each row, which may be problematic for tables with thousands rows.

```php
$query = $db->simple_select("moderatorlog", "*");
while ($log = $db->fetch_array($query))
{
    // Do stuff
}
```

This would be better. However, there's still room for improvement in the update_query() part, which is also ran for each row. I'm a bit busy at the moment so I can't write down the complete function I'm afraid.

Reply:
To confirm you'd like me to switch to a while loop and update every row that way? Keep in mind that regardless of how we do it every row will still be read and if applicable updated. However it could be adjusted to only update if the row contains a slash which might be the code slightly more efficient although then that ends up with another check on each row which could slow things down even further. Thoughts?

Reply:
See @euantorano's reply below :)